collector/pipelinerun.go

```diff
@@ -2,27 +2,57 @@ package collector
 
 import (
 	"context"
-	"github.com/go-kit/log/level"
+	"fmt"
+
 	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
-	"os"
 
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/metrics"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-func SetupPipelineRunCachingClient(mgr ctrl.Manager) {
-	// we do not set up a reconciler to get a caching client, but we make one list
-	// call up top to get the ball rolling; confirmed via break points in a debugger that
-	// k8s methods for creating a shared informer and watch are called
-	prList := v1beta1.PipelineRunList{}
-	opts := &client.ListOptions{Namespace: metav1.NamespaceAll}
-	mgr.GetClient().List(context.Background(), &prList, opts)
-	psCollector, err := NewCollector(logger, mgr.GetClient())
+type ReconcilePipelineRun struct {
+	client        client.Client
+	scheme        *runtime.Scheme
+	eventRecorder record.EventRecorder
+	psCollector   *PipelineServiceCollector
+}
+
+func SetupPipelineRunCachingClient(mgr ctrl.Manager) error {
```

Comment:
naming nit - this is setting a controller, not a client?

```suggestion
func SetupPipelineRunScheduleDurationController(mgr ctrl.Manager) error {
```

Reply:
ok will do in next push